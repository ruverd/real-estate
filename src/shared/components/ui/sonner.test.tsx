import { render } from "@testing-library/react";
import { vi } from "vitest";

import { Toaster } from "./sonner";

// Mock next-themes
vi.mock("next-themes", () => ({
  useTheme: () => ({ theme: "light" }),
}));

// Mock sonner
vi.mock("sonner", () => ({
  Toaster: ({ theme, className, toastOptions }) => (
    <div data-testid="mock-sonner" data-theme={theme} className={className}>
      <div data-testid="toast-options" data-options={JSON.stringify(toastOptions)}></div>
    </div>
  ),
}));

describe("Sonner", () => {
  it("should render toaster with theme from next-themes", () => {
    const { getByTestId } = render(<Toaster />);
    
    const sonner = getByTestId("mock-sonner");
    expect(sonner).toHaveAttribute("data-theme", "light");
    expect(sonner).toHaveClass("toaster");
    expect(sonner).toHaveClass("group");
  });

  it("should pass toast options to sonner", () => {
    const { getByTestId } = render(<Toaster />);
    
    const toastOptions = getByTestId("toast-options");
    const options = JSON.parse(toastOptions.getAttribute("data-options") || "{}");
    
    expect(options.classNames).toBeDefined();
    expect(options.classNames.toast).toContain("group toast");
    expect(options.classNames.description).toContain("group-[.toast]:text-muted-foreground");
    expect(options.classNames.actionButton).toContain("group-[.toast]:bg-primary");
    expect(options.classNames.cancelButton).toContain("group-[.toast]:bg-muted");
  });

  it("should pass custom props to sonner", () => {
    const { getByTestId } = render(<Toaster position="top-left" closeButton />);
    
    // In a real test, we would check if these props were passed to the Sonner component
    // Since we're using a mock, we're just checking that the component renders
    expect(getByTestId("mock-sonner")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<Toaster />);
    expect(container).toMatchSnapshot();
  });
});
