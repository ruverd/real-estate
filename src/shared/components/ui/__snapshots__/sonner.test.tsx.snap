// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sonner > should match snapshot 1`] = `
<div>
  <div
    class="toaster group"
    data-testid="mock-sonner"
    data-theme="light"
  >
    <div
      data-options="{"classNames":{"toast":"group toast group-[.toaster]:bg-background group-[.toaster]:text-foreground group-[.toaster]:border-border group-[.toaster]:shadow-lg","description":"group-[.toast]:text-muted-foreground","actionButton":"group-[.toast]:bg-primary group-[.toast]:text-primary-foreground font-medium","cancelButton":"group-[.toast]:bg-muted group-[.toast]:text-muted-foreground font-medium"}}"
      data-testid="toast-options"
    />
  </div>
</div>
`;
